### PR TITLE
Fix #475

### LINKED ISSUE #475: Ruffle ignores content inside iframes
Ruffle will always ignore any Flash content inside iframe tags. Though this makes sense for the self-hosted (JS) build, this is more unexpected from the extension.

Attached a sample:
- Add a JS build to "ruffle/", then run "web.html", notice Ruffle does not convert the file.
- Run "extension.html" with the extension, notice the same behavior.
- "workaround.html" is an attempt to fix the issue on the JS build (actually it just duplicates the Ruffle configuration on the parent page and appends it to each iframe).

[iframeruffle.zip](https://github.com/ruffle-rs/ruffle/files/4389981/iframeruffle.zip)

Also, I haven't tried, but I would expect the same behavior with the old frameset / frame tags.

### PR DESCRIPTION
**Pull request by Justin-CB**
This fix still needs testing, and has 2 changes: an addition to the ruffle extension's manifest.json to load it into all frames, and 2 new polyfills for the hosted version.

### CODE REVIEW

**Comment by Justin-CB** [comment]
Mainly the hosted version needs testing because the extension change is relatively minor.

File: web/js-src/polyfills.js [hunk 1]
```diff
@@ -60,11 +76,90 @@ function falsify_plugin_detection() {
     install_plugin(FLASH_PLUGIN);
 }
 
+function polyfill_frames(depth)
+{
+    var i;
+    var current_frame;
+    var frame_document;
+    var script;
```
Comment (Herschel) [inline web/js-src/polyfills.js:84]: Let's try to use `let` instead of `var`

File: web/js-src/polyfills.js [hunk 2]
```diff
@@ -14,6 +28,8 @@ let embeds = null;
  * keep native objects out of the DOM, and thus out of JavaScript's grubby
  * little hands, but only if we load first.
  */
+var objects;
+var embeds;
```
Comment (Herschel) [inline web/js-src/polyfills.js:32]: Let's try to use `let` instead of `var`

File: web/js-src/polyfills.js [hunk 3]
```diff
@@ -60,11 +76,90 @@ function falsify_plugin_detection() {
...
+    /* Basically the same as the listener for dynamic embeds. */
+    let nodesAdded = mutationsList.some(mutation => mutation.addedNodes.length > 0);
+    if (nodesAdded) {
+        polyfill_frames(window);
+        install_dynamic_frame_polyfill_on_subframes(window, ruffle_frame_listener);
+    }
+}
+
+function install_dynamic_frame_polyfill_on_subframes(depth, observer)
+{
```
Comment (Herschel) [inline web/js-src/polyfills.js:127]: Nit: Put `{` on the same line to match style of the rest of the code

File: web/js-src/polyfills.js [hunk 4]
```diff
@@ -1,11 +1,25 @@
...
+}
 
+/* Save current RufflePlayer settings and *
+ * to propagate to any frames on the page *
+ * if we chuse to polyfill them.          */
+var top_level_ruffle_player=window.RufflePlayer;
+var ruffle_script_src=public_path({}, "ruffle.js");
+if (window.RufflePlayer.config) {
+    ruffle_script_src=public_path(window.RufflePlayer.config, "ruffle.js");
+}
```
Comment (Herschel) [inline web/js-src/polyfills.js:17]: I think this can just be `document.currentScript.src`

File: web/extension/build/js/lv0.js [hunk 5]
```diff
@@ -39,4 +39,4 @@
                 console.error("Critical error loading Ruffle into page")
             }
         }
-    }());
\ No newline at end of file
+    }());
```
Comment (Herschel) [inline web/extension/build/js/lv0.js]: Please avoid unrelated changes in commits

Comment (Justin-CB) [reply web/extension/build/js/lv0.js]: I think I edited lv0.js when debugging, and git added a new line to the end of the file(For some reason the javascript sources were missing newlines at the ends of the files).

**Comment by Justin-CB** [comment]
I didn't mean to close this. I just rebased my branch for a nicer history.

**Comment by Justin-CB** [comment]
I addressed all the coding style issues & bugfixes you mentioned. Of course, the CORS discussion is still up in the air.

**Comment by Justin-CB** [comment]
Static & dynamic frames & frame navigation are all working if it's served from a server, but RufflePlayer.config isn't being populated properly to subframes for some reason
[iframeruffle.tar.gz](https://github.com/ruffle-rs/ruffle/files/4500629/iframeruffle.tar.gz)
.

**Comment by Justin-CB** [comment]
Now the frame polyfill isn't being applied in frames for whatever reason. Whether or not there is code to iterate over frames' frames doesn't appear to make it work or not work.

**Comment by Justin-CB** [comment]
OK, everything is working now.

**Comment by Justin-CB** [comment]
I ran a rebase against ruffle-rs/ruffle master, & now it's failing circleci with a complaint about desktop/src/navigator.rs

**Comment by Justin-CB** [comment]
OK, so it looks like the main ruffle repository's latest commits are having the same error, so it isn't my fault.

**Comment by Herschel** [comment]
Thanks! The only issues I see now:
 * This should probably work for oldschool `<frame>` as well.
* It feels a little redundant to have both `static-frames` and `dynamic-frames`; let's just have `frames` and base the "dynamicness" off of whether `dynamic-content` is also set. This probably involves passing the list of polyfills or a boolean to the polyfill function.

**Comment by Herschel** [comment]
Here's an example of classic frames:
[frame.zip](https://github.com/ruffle-rs/ruffle/files/4536549/frame.zip)

**Comment by Justin-CB** [comment]
It was a simple change to make it work(it took 2 commits, tho, because I made a typo).

**Comment by Herschel** [comment]
Thank you! Because we already have `dynamic-content`, I'd like to merge `static-frames` and `dynamic-frames` and only have a `frames` polyfill setting, with the dynamicness-based on whether `dynamic-content` is set.

**Comment by Justin-CB** [comment]
Any other change requests?

**Comment by Herschel** [comment]
Thank you!

**Pull request merged**